### PR TITLE
core-components: show arrow heads in dependency graph

### LINKED ISSUE #21587: 🚀 Feature: Dependency Graph Arrows
### 🔖 Feature description

It will be great to have arrows on the dependency relations graph. It makes relations easier to understand.

### 🎤 Context

As I see arrows were removed in this PR https://github.com/backstage/backstage/pull/13673. Is it possible to enable it back with some props?

### ✌️ Possible Implementation

_No response_

### 👀 Have you spent some time to check if this feature request has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

No, I don't have time to work on this right now

### PR DESCRIPTION
**Pull request by mhitkaohp**
## Hey, I just made a Pull Request!

Show arrow heads in the dependency graph. Fixes #21587.

The dependency graph component used to show arrow heads before, this reverts part of https://github.com/backstage/backstage/pull/13673/, reusing `ARROW_MARKER_ID` that has been unused since then. I slightly increased the stroke width as well since that has influence on the size of the arrow heads.

#### Before and after: A depends on B, B depends on C, C depends on A (not simplified, with merged relations)
before
after

#### Before and after: artist engagement portal from Backstage examples (not simplified, without merged relations)
before
after

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [x] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: packages/core-components/src/components/DependencyGraph/Edge.tsx
Comment (awanlin) [inline packages/core-components/src/components/DependencyGraph/Edge.tsx:129]: While I really like this change I do think it should be opt-in given that it might be possible for some existing Adopters that this creates confusion as notes by @freben here: https://github.com/backstage/backstage/issues/21587#issuecomment-1833698235

**Review by awanlin — CHANGES_REQUESTED** [review]
Thanks for this @mhitkaohp, just left one comment but over all the makes sense to me and looks good 👍

File: packages/core-components/src/components/DependencyGraph/Edge.tsx
Comment (mhitkaohp) [reply packages/core-components/src/components/DependencyGraph/Edge.tsx:129]: @awanlin yep, I saw that comment and it wasn't convincing, did you see my answer there - https://github.com/backstage/backstage/issues/21587#issuecomment-1889445662? It's a pity that the removal of the arrow heads (as part of https://github.com/backstage/backstage/pull/13673) wasn't made opt-in (or at least opt-out), I'd argue that it's the current behaviour that creates confusion, I tried to illustrate that with the screenshots in the PR description.

If the worry is about existing adopters, wouldn't it be acceptable to make it opt-out instead? I believe that changing the default behaviour will improve the understandability of any non-trivial diagram in Backstage, so I'd love to have a feeling at least that I managed to explain myself, even if at the end I won't convince you or @freben.

Comment (awanlin) [reply packages/core-components/src/components/DependencyGraph/Edge.tsx:129]: Yes, I did see the various comments. We generally lean to opt-in for these changes and use the changeset to note how to enable them in the past. I'm also basing my thinking on the data I have - I'm very active on Discord helping people and the arrows are rarely a problem that has ben bought up from that group of people and the issue you logged doesn't have many comments or interactions. This leans my thinking again to making this opt-in.

To be clear I think this is an improvement and will be helpful. Also, I'm not the final say on this, that I defer to the Maintainers like @freben to decide on and will be fine with whatever they decide. 👍

Comment (mhitkaohp) [reply packages/core-components/src/components/DependencyGraph/Edge.tsx:129]: Thanks for the explanation and your time. I prefer simpler solutions since they are easier to maintain but I understand that flexibility might be more important for a project like Backstage, so it shall be configurable. I'd still advocate for making it at least opt-out since there are cases when the current rendering is misleading and I don't really see how it could harm the existing adopters. Even the videos in the catalog graph documentation contain arrow heads: https://github.com/backstage/backstage/blob/master/plugins/catalog-graph/README.md. Nevertheless even having it opt-in scratches my itch, so I'll push a new iteration soon.

I'd be happy to change to opt-out if there is interest of course :wink:.

File: .changeset/forty-beers-sniff.md
Comment (awanlin) [inline .changeset/forty-beers-sniff.md:6]: Can you include just a small code sample of how this gets used here, please? These changesets get rolled up in the the changelog for each release so having a basic example here means if this is a feature you want you can easily implement it.

**Review by awanlin — CHANGES_REQUESTED** [review]

File: .changeset/forty-beers-sniff.md
Comment (mhitkaohp) [reply .changeset/forty-beers-sniff.md:6]: Right, on it. Let me know if you'd also like to adjust the wording.

File: .changeset/forty-beers-sniff.md
Comment (mhitkaohp) [inline .changeset/forty-beers-sniff.md:12]: Hmm, I took this from the App.tsx in the dev app, but I see now that it would be better to use the one in the create-app template which is much simpler out of the box, one more iteration.

Comment (awanlin) [reply .changeset/forty-beers-sniff.md:12]: Yeah, super sorry, all I was looking for was a simple one-liner to show the new prop as a hint. If we can simplify this that would be great. Again, sorry, really not trying to make this a painful process :(

Comment (mhitkaohp) [reply .changeset/forty-beers-sniff.md]: Simplified

Comment (mhitkaohp) [reply .changeset/forty-beers-sniff.md]: No worries, thanks for fast feedback, makes it much less painful.

**Review by awanlin — APPROVED** [review]
Awesome! Just need review/approval from the Maintainers now 🚀

**Pull request merged**
